core/src/page_id.rs

```diff
@@ -0,0 +1,170 @@
+//! TODO: add docs
+
+use crate::{page::DEPTH, trie::KeyPath};
+use bitvec::{prelude::*, slice::ChunksExact};
+
+pub type PageId = [u8; 32];
+
+pub const ROOT_PAGE_ID: [u8; 32] = [0; 32];
+
+const MAX_CHILD_INDEX: u8 = (1 << DEPTH) - 1;
+
+pub fn child_page_id(mut page_id: PageId, child_index: u8) -> PageId {
+    // TODO: proper error handling
+    assert!(child_index & 0b11000000 == 0);
```

Comment:
this is less readable than `assert!(child_index < MAX_CHILD_INDEX)`. We should not be making the reader spend cycles on things that can be easy unless there is a driving performance need.

Reply:
also, generally I consider `assert` or panicking bad practice over error handling, but there is a TODO which should include the two failure cases:
  1. child index out of bounds
  2. page id overflow

Reply:
thanks, yes `assert!(child_index <= MAX_CHILD_INDEX)` is for sure more readable. I will implement proper error handling though!